File: Checks/PWD010/example-omp.c
// PWD010: Incorrect sharing in parallel region

void example(int **result, unsigned rows, unsigned cols) {
  int i, j;

  // j is implicitly shared and it should be private!
  // NOT-PWR004: iterator variables aren't specified to showcase the issue
  #pragma omp parallel for shared(result)
  for (i = 0; i < rows; i++) {
    for (j = 0; j < cols; j++) {
      result[i][j] = 0;
    }
  }
}

File: Checks/PWD010/example-omp.f90
! PWD010: Incorrect sharing in parallel region

subroutine example(result)
  integer, intent(out) :: result(:, :)
  integer :: i, j

  ! NOT-PWR004: iterator variables aren't specified to showcase the issue
  !$omp parallel do shared(result)
  do j = 1, size(result, 2)
    do i = 1, size(result, 1)
      result(i, j) = 0
    end do
  end do
end subroutine example
